packages/aws-cdk-lib/aws-lambda/test/code.test.ts

```diff
@@ -615,6 +616,96 @@ describe('code', () => {
       expect(cpMock).toHaveBeenCalledWith('/my/image/path/.', undefined);
     });
   });
+
+  describe('lambda.Code.fromBucket', () => {
+    test('fromBucket warns when no objectVersion is set', () => {
+      // given
+      const app = new cdk.App();
+      const stack = new cdk.Stack(app, 'Stack');
+      const bucket = new s3.Bucket(stack, 'Bucket');
+
+      // when
+      new lambda.Function(stack, 'Fn', {
+        code: lambda.Code.fromBucket(bucket, 'Object'),
+        handler: 'index.handler',
+        runtime: lambda.Runtime.NODEJS_LATEST,
+      });
+
+      // then
+      Annotations.fromStack(stack).hasWarning(
+        '/Stack/Bucket',
+        'objectVersion is not defined for S3Code.fromBucket(). The lambda will not be updated automatically if the code in the bucket is updated. ' +
+        'This is because CDK/Cloudformation does not track changes on the source S3 Bucket. It is recommended to either use S3Code.fromAsset() instead or set objectVersion. ' +
+        '[ack: @aws-cdk/aws-lambda:codeFromBucketObjectVersionNotSpecified]',
+      );
+    });
+
+    test('fromBucket does not warn when an objectVersion is set', () => {
+      // given
+      const app = new cdk.App();
+      const stack = new cdk.Stack(app, 'Stack');
+      const bucket = new s3.Bucket(stack, 'Bucket');
+
+      // when
+      new lambda.Function(stack, 'Fn', {
+        code: lambda.Code.fromBucket(bucket, 'Object', 'v1'),
+        handler: 'index.handler',
+        runtime: lambda.Runtime.NODEJS_LATEST,
+      });
+
+      // then
+      Annotations.fromStack(stack).hasNoWarning(
+        '/Stack/Bucket',
+        'objectVersion is not defined for S3Code.fromBucket(). The lambda will not be updated automatically if the code in the bucket is updated. ' +
+        'This is because CDK/Cloudformation does not track changes on the source S3 Bucket. It is recommended to either use S3Code.fromAsset() instead or set objectVersion. ' +
+        '[ack: @aws-cdk/aws-lambda:codeFromBucketObjectVersionNotSpecified]'
```

Comment:
```suggestion
        '[ack: @aws-cdk/aws-lambda:codeFromBucketObjectVersionNotSpecified]',
```

this fixes the lint failure